packages/paste-core/components/toast/__tests__/index.spec.tsx

```diff
@@ -56,6 +56,92 @@ describe('Toast', () => {
     });
   });
 
+  describe('i18n', () => {
+    it('should have default dismiss button text', () => {
+      render(
+        <Toast variant="neutral" onDismiss={onDismissMock}>
+          This is a toast
+        </Toast>
+      );
+      const dismissButton = screen.getByRole('button', {name: 'Dismiss toast'});
+      expect(dismissButton).toBeDefined();
+    });
+
+    it('should use i18nDismissLabel for dismiss button text', () => {
+      render(
+        <Toast variant="neutral" i18nDismissLabel="Cerrar notificacion" onDismiss={onDismissMock}>
+          Esta es una notificacion
+        </Toast>
+      );
+      const dismissButton = screen.getByRole('button', {name: 'Cerrar notificacion'});
+      expect(dismissButton).toBeDefined();
+    });
+
+    it('should have default error variant icon text', () => {
+      render(<Toast variant="error">This is a toast</Toast>);
+      const iconText = screen.getByText('(error)');
+      expect(iconText).toBeDefined();
+    });
+
+    it('should have default neutral variant icon text', () => {
+      render(<Toast variant="neutral">This is a toast</Toast>);
+      const iconText = screen.getByText('(information)');
+      expect(iconText).toBeDefined();
+    });
+
+    it('should have default success variant icon text', () => {
+      render(<Toast variant="success">This is a toast</Toast>);
+      const iconText = screen.getByText('(success)');
+      expect(iconText).toBeDefined();
+    });
+
+    it('should have default warning variant icon text', () => {
+      render(<Toast variant="warning">This is a toast</Toast>);
+      const iconText = screen.getByText('(warning)');
+      expect(iconText).toBeDefined();
+    });
+
+    it('should use i18n prop for error variant icon text', () => {
+      render(
+        <Toast variant="error" i18nErrorLabel="(erreur)">
+          This is a toast
+        </Toast>
+      );
+      const iconText = screen.getByText('(erreur)');
+      expect(iconText).toBeDefined();
+    });
+
+    it('should use i18n prop for neutral variant icon text', () => {
+      render(
+        <Toast variant="neutral" i18nNeutralLabel="(informacion)">
+          This is a toast
+        </Toast>
+      );
+      const iconText = screen.getByText('(informacion)');
+      expect(iconText).toBeDefined();
```

Comment:
thought(non-blocking): I actually have no good solution for this, it's just an observation. 

It terms of a test I wonder how effective this will be long term. We essentially test, "does _an_ element with this string exist?" Which proves the prop works, but might not prove the prop does what it's semantically meant to do.  To explain that further, I could theoretically move the screen reader only text to anywhere in the toast composition and the test would still pass, but the icon would no longer with described with a translated string.

Reply:
Further expansion, the dismiss button tests are really good examples. It tests that there is a button, it is a dismiss button and it is translated.

There really isn't a good "icon" specific alternative

Reply:
For sure, I think the only way we can check that specifically is if we add a `data` attribute somewhere and then checking if the text exists within the element with that attribute.

It could go on the `MediaFigure` that wraps the icon & text... or we could go back to using the `title` attribute on the icon and use the `data-paste-element` attribute on the icon... or add a data attribute on the `ScreenReaderOnly` element..,

Is there a reason why we decided to use `ScreenReaderOnly` instead of the `title` attribute on the icon for i18n?

Reply:
I'll be merging this as the conversation continues, but I appreciate everyone's ideas and input, especially since it's bound to come up again. Thank you!